src/main/kotlin/compiler/Parser.kt

```diff
@@ -273,10 +274,48 @@ class Parser(inputTokens: List<Token>) {
     // Parse any expression we find next.
     // Start with the lowest precedence operation, passing down to look for higher precedence operations.
     private fun pExpression(): N_EXPR? {
+        val next = this::pWhenExpr
```

Comment:
Here (at the top of the expression stack) we parse the expression version of 'when'.